### PR TITLE
[NS11,NS12,NS13]: feat/network-api-v2

### PR DESCRIPTION
**Pull request by fabioespinosa**
- Adds endpoint to download links for a :fileId inside a :bucket

### CODE REVIEW

File: lib/server/routes/buckets.js [hunk 1]
```diff
@@ -1413,6 +1416,332 @@ BucketsRouter.prototype.getStorageUsage = function (req, res) {
...
+              if (err) {
+                return nextShardId(err);
+              }
+
+              if (!contact) {
+                return nextShardId(new errors.NotFoundError('Contact not found'));
+              }
+
+              const { address, port } = contact;
+              const farmerUrl = `http://${address}:${port}/download/link/${hash}`;
```
Comment (fabioespinosa) [inline lib/server/routes/buckets.js]: shouldn't this be the uuid?

Comment (sg-gs) [reply lib/server/routes/buckets.js]: yes, should be

Comment (sg-gs) [reply lib/server/routes/buckets.js]: ping me when it is changed to proceed with this

File: lib/server/routes/buckets.js [hunk 2]
```diff
@@ -1413,6 +1416,361 @@ BucketsRouter.prototype.getStorageUsage = function (req, res) {
...
+
+          async.eachLimit(
+            uploads,
+            1,
+            (upload, cb) => {
+              upload.remove((err) => {
+                if (err) {
+                  log.error('Error deleting Upload model with id: %s, error: %s', upload._id, err.message);
+
+                  return;
```
Comment (sg-gs) [inline lib/server/routes/buckets.js]: use `cb(err)` and print the log in the eachLimit callback. Thus, we can avoid letting `eachLimit` stalled

**Review by sg-gs — CHANGES_REQUESTED** [review]
Great job @fabioespinosa. Excited to try this new API on the new upload/download

**Comment by sg-gs** [comment]
Also, resolve conflicts with master before merging

File: lib/server/routes/buckets.js (hunk 2 above)
Comment (fabioespinosa) [reply lib/server/routes/buckets.js]: Fixed

**Comment by fabioespinosa** [comment]
> Also, resolve conflicts with master before merging

Should be ok now :)

**Comment by sg-gs** [comment]
Seems that the `uuid-validate` package is not added as a dependency on package.json. Also, is preferred to use `uuid` package always when possible, as is RFC4122 compliant. In this case `uuid` [has a validation method](https://www.npmjs.com/package/uuid#uuidvalidatestr)

**Review by sg-gs — CHANGES_REQUESTED** [review]
Seems that the `uuid-validate` package is not added as a dependency on package.json. Also, is preferred to use `uuid` package always when possible, as is RFC4122 compliant. In this case `uuid` [has a validation method](https://www.npmjs.com/package/uuid#uuidvalidatestr)

**Pull request merged**